### PR TITLE
fix:changed celery health-check endpoint

### LINKED ISSUE #6471: Implement docker healthcheck and fix endpoint
Currently, the health check endpoint shows that celery is not up. Which is untrue
https://api.eventyay.com/health-check

Besides, the presence or absence of celery does not signify the health of the web container. Hence, only DB and migrations should be checked by default and celery should have another endpoint for itself for checking the celery container

The second part of the issue is to add healthcheck command to Dockerfile so that 
1. DockerHub can automatically test to see if it runs or not (will most likely fail as it depends on PSQL)
2. Docker Compose can restart brain dead containers - live containers no longer fit for serving requests

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #6471 

#### Short description of what this resolves:
Removed celery check from health-check of app, added a new end point with queryparam for celery check

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by codedsun** [comment]
@iamareebjamal Currently I am getting error for celery endpoint 

```

File "/usr/local/lib/python3.7/site-packages/healthcheck/healthcheck.py", line 139, in run_check
        elapsed_time = end_time - start_time
        # Reduce to 6 decimal points to have consistency with timestamp
        elapsed_time = float('{:.6f}'.format(elapsed_time))
 
        if not passed:
            msg = 'Health check "{}" failed with output "{}"'.format(checker.__name__, output)
            logger.error(msg)
 
        timestamp = time.time()
        if passed:
            expires = timestamp + self.success_ttl
AttributeError: 'tuple' object has no attribute '__name__'

```

File: app/__init__.py [hunk 1]
```diff
@@ -229,12 +229,13 @@ def make_celery(app=None):
...
 # Health-check
-health = HealthCheck(current_app, "/health-check")
-health.add_check(health_check_celery)
+health = HealthCheck()
+health_dependencies = HealthCheck()
 health.add_check(health_check_db)
 with current_app.app_context():
     current_app.config['MIGRATION_STATUS'] = check_migrations()
 health.add_check(health_check_migrations)
+health_dependencies.add_check(health_check_celery())
```
Comment (iamareebjamal) [inline app/__init__.py]: See what was written previously and what you have written

Comment (codedsun) [reply app/__init__.py]: @iamareebjamal Previously it was written 
`health.add_check(health_check_celery)` But how do we add to same instance of health-check when we want only celery result at a seperate endpoint.

Comment (codedsun) [reply app/__init__.py]: I have made a different instance to get the result of celery

Comment (iamareebjamal) [reply app/__init__.py]: Again you are looking at a completely different thing than what I am trying to point out

```diff
-health.add_check(health_check_celery)
+health_dependencies.add_check(health_check_celery())
```

Comment (iamareebjamal) [reply app/__init__.py]: Can you see any other difference than health and health_dependencies?

Comment (codedsun) [reply app/__init__.py]: @iamareebjamal Please read the comment sir. I am looking at the same line

Comment (codedsun) [reply app/__init__.py]:
```
health.add_check(health_check_celery)
```

Comment (codedsun) [reply app/__init__.py]: @iamareebjamal I am at the same point and direction you said. Also I wrote why I did this, please read.

Comment (codedsun) [reply app/__init__.py]: Yes I saw. Sorry

Comment (codedsun) [reply app/__init__.py]: But it's a function, in python dont they have ( ) braces when calling a function. Dumb me. Apologies

**Comment by codedsun** [comment]
@iamareebjamal Done. thanks

File: app/__init__.py (hunk 1 above)
Comment (codedsun) [reply app/__init__.py]: Writing kotlin and java daily had made me done this :( Will not repeat again. Thanks

**Comment by codedsun** [comment]
Is any task pending in this, if not. Please check. Thankyou

**Comment by codedsun** [comment]
@iamareebjamal To restart brain dead containers, we have to use docker swarm?

https://www.katacoda.com/courses/docker-orchestration/healthcheck

I have added the health check command in dockerfile

File: Dockerfile [hunk 2]
```diff
@@ -26,4 +26,5 @@ WORKDIR /data/app
 ADD . .
 
 EXPOSE 8080
+HEALTHCHECK CMD curl --fail http://localhost:5000/health-check || exit 1
```
Comment (iamareebjamal) [inline Dockerfile]: It should check proper healthcheck. Normal for API container and celery for celery container

Comment (codedsun) [reply Dockerfile]: Okay that means I have to add for respective containers in docker-compose file. THanks

Comment (iamareebjamal) [reply Dockerfile]: No

Comment (codedsun) [reply Dockerfile]: then two healthcheck command in this DOCKERFILE? like this? 

```
HEALTHCHECK CMD curl --fail http://localhost:5000/health-check || exit 1
HEALTHCHECK CMD curl --fail http://localhost:5000/health-check?q=celery || exit 1
```

Comment (iamareebjamal) [reply Dockerfile]: There can't be 2 HEALTHCHECKs

Comment (codedsun) [reply Dockerfile]: @iamareebjamal Their can be only 1 healthcheck command in dockerfile but we have 2 different URLs for healthcheck(one for app, other for celery). Also we dont' have to add the healthcheck in docker-compose file for respective containers. Then how can I do it? Can you help me with this. 

Also, To restart brain dead containers, we have to use docker swarm?

Comment (iamareebjamal) [reply Dockerfile]: See container start shell file

Comment (codedsun) [reply Dockerfile]: @iamareebjamal Got it have to access the deployment variable in dockerfile and based on value of it add healthcheck? correct?

**Comment by codedsun** [comment]
@iamareebjamal May be I should correct this https://github.com/fossasia/open-event-server/pull/6599 and it will get me the args directly from the docker-compose file to the dockerfile and based on it I will add the the healthcheck. Should I follow this way and also am I correct?

**Pull request closed without merging**